### PR TITLE
Run the tests in a xenial LXC.

### PR DESCRIPTION
**Pull request by elopio**
Fixes LP: [#1532213](https://bugs.launchpad.net/snapcraft/+bug/1532213)

### CODE REVIEW

**Comment by elopio** [comment]
The examples tests are failing because of https://bugs.launchpad.net/ubuntu/+source/make-dfsg/+bug/1536727.
Our idea has been always to run them in Jenkins because there we can access the daily snappy installed on a cloud instance and run also the internal and external commands. The job for that is ready, but pending a Jenkins redeploy. In the meantime if needed, I will be running the example suite locally.

File: .travis.yml [hunk 1]
```diff
@@ -10,21 +10,33 @@ env:
   # Encrypted Coveralls token. The secret token for your repository can be found at the
   # bottom of your repository's page on Coveralls. To encrypt it, install the travis gem
   # and run:
-  #     travis encrypt -r ubuntu-core/snappy COVERALLS_TOKEN=$secret_token
-  - secure: JJEh1YfzNJGb1//ca447JtRunHmNXsVDGy5fS882/RQV3XtS9mCaSS+PZF4AbVABSiDw96mvwv05KmhdgV575y0NVHuRPd3rJBku9HOmhOKfCykn5tXgVJUEgeW5HIPusltRkXJUxnm48l+RfHHJU1h6iJnpGjvwxygeQljM4d22+G493ymvv4KZJSnbtuDbmhWvnZtdV2cCFeZ+EWqLgyWaFQmMgeMkYYcY3Cy19d++WquLs4woN9S9szrXJQlIv3xuZ8XNLbWmHoe6uD4y1Fn7YYfRgz861KqusxcGO9OyCh1S6uLgDuETS4Uw2zBg77vv59jSJieEfcY8jQtFE65axot/90jwhp5IV0tIpd9O/AMZXIGqKk44RU+Sj9gDbPbBIhj9dew3BTwSuOcNITSg1a4sOkxHGyIj2NozCkgFqPq0d4RM7xRrOYfnmjP1sJ+tEH4XMbs4gNZ/abccJXeUx6i5qWhDdnix132C62vvl+ItnsfMtkXQ34hluYxoab+rS0Zz3MM/ENqHySvU/R2n7+WTVXHyoK55HvExX7Y+McqPU2uVsHDabmDmQfcYom9nDY1OTBM3JbTOuqtIrzSIlFYdogyttXrTsmSue8TAoWWnHirKdkzeskIs5ZbI/pfGV3eRol+9Mn0MtmaoLp1VrAtSiZYwPRdcIGBDNMQ=
+  #     travis encrypt -r ubuntu-core/snappy COVERALLS_REPO_TOKEN=$secret_token
```
Comment (kyrofa) [inline .travis.yml]: Ah, I didn't know they changed this.

File: .travis.yml [hunk 2]
```diff
@@ -10,21 +10,33 @@ env:
...
   - TEST_SUITE=integration
-  - TEST_SUITE="examples --skip-install"
 
 install:
-  # PPA for plainbox.
-  - sudo apt-add-repository -y ppa:hardware-certification/public
+  - sudo apt-add-repository -y ppa:ubuntu-lxc/lxd-stable
   - sudo apt-get update -qq
-  - sudo apt-get install -qq build-essential squashfs-tools dpkg-dev plainbox pyflakes python3-apt python3-docopt python3-coverage python3-fixtures python3-jsonschema python3-mccabe python3-pep8 python3-requests python3-testscenarios python3-testtools python3-yaml python3-lxml
+  - sudo apt-get -o Dpkg::Options::="--force-confnew" -y dist-upgrade || true
```
Comment (kyrofa) [inline .travis.yml]: Yuck.

**Comment by kyrofa** [comment]
Looks good to me.

File: .travis.yml (hunk 2 above)
Comment (elopio) [inline .travis.yml]: indeed. Most of the ugliness comes from travis preinstalling a lot of stuff in the trusty vms. The problem now is with postgresql, but could happen with many of the other things if they fail to upgrade or start. We need none of them, so it's generally safe to ignore the errors. And if there's an error that affects us, the test will fail below.

**Comment by elopio** [comment]
@sergiusens waiting for you +0.02 here.

File: snapcraft/plugins/catkin.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
-# Copyright © 2015 Canonical Ltd
+# Copyright (C) 2015, 2016 Canonical Ltd
```
Comment (sergiusens) [inline snapcraft/plugins/catkin.py]: Are you afraid of utf-8? :smile:

File: .travis.yml [hunk 4]
```diff
@@ -10,21 +10,33 @@ env:
...
+  - sudo apt-get install -qq lxd lxcfs || true
+before_script:
+  - sudo lxd-images import ubuntu xenial --stream daily --alias ubuntu
+  - sudo lxc launch ubuntu xenial -c security.privileged=true
```
Comment (sergiusens) [inline .travis.yml]: aha! *privileged*

**Comment by sergiusens** [comment]
:+1:

File: snapcraft/plugins/catkin.py (hunk 3 above)
Comment (elopio) [inline snapcraft/plugins/catkin.py]: of course :)

File: .travis.yml (hunk 4 above)
Comment (elopio) [inline .travis.yml]: yup, but it didn't make any difference in the end. I was trying to use it to get rid of the chmod 777.
I still want to get rid of it, but I prefer to keep trying in a different PR.

**Comment by elopio** [comment]
I'm so very happy that it finally landed. :beers:

**Comment by kyrofa** [comment]
Hey! Congratulations @elopio!
